pkg/sql/codegen/tensorflow/codegen.go

```diff
@@ -159,9 +157,19 @@ func IsKerasModel(estimator string) (bool, string) {
 	return false, fmt.Sprintf("tf.estimator.%s", estimator)
 }
 
+func validateAttributes(trainIR *ir.TrainClause) error {
+	modelAttr := attribute.NewDictionary(trainIR.Estimator, "model.")
+	for name, attr := range modelAttr {
+		if strings.HasSuffix(name, "optimizer") {
```

Comment:
I think this is kind of temporary. The `optimizer` argument in Tensorflow Estimators can be a string as well as an `Optimizer` object, we need also pass things like `learning_rate`, `learning_rate_decay` etc. to construct the optimizer object.

Seems it's better to use a struct:

```
type Optimizer struct {
    Name string
    params map[string]interface{}
}
```
To store the construction details of the optimizer object. If `params` is empty, then pass the optimizer name as string to the Estimator, else construct an object and pass it to Estimator.

And, it's useful to update `python/sqlflow_submitters/tensorflow/train.py` to have an argument `optimizer` to accept it.

Reply:
For reference: https://www.tensorflow.org/api_docs/python/tf/keras/optimizers

Reply:
> I think this is kind of temporary. The `optimizer` argument in Tensorflow Estimators can be a string as well as an `Optimizer` object, we need also pass things like `learning_rate`, `learning_rate_decay` etc. to construct the optimizer object.
> 
> Seems it's better to use a struct:
> 
> ```
> type Optimizer struct {
>     Name string
>     params map[string]interface{}
> }
> ```
> 
> To store the construction details of the optimizer object. If `params` is empty, then pass the optimizer name as string to the Estimator, else construct an object and pass it to Estimator.
> 
> And, it's useful to update `python/sqlflow_submitters/tensorflow/train.py` to have an argument `optimizer` to accept it.

Yes, this is a transitional plan. To allow constructing an `Optimizer` in SQL statement requires a more comprehensive design on which I'm still working.

Reply:
> To allow constructing an Optimizer in SQL statement requires a more comprehensive design on which I'm still working.

Maybe something like the following?

```
WITH
    optimizer.name = "adam"
    optimizer.param.lr = 0.1
```